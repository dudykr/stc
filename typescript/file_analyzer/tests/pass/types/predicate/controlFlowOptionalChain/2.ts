declare const x: string | number;
declare const o2: { f(x: any): x is number; } | undefined;
if (o2?.f(x)) {
    x; // number
    o2.f; // (x: any) => x is number
    o2?.f;
    o2?.f(x);
}
else {
    x;
    o2;
    o2?.f;
    o2.f;
}
x;
o2;
o2?.f;
o2.f;

export { }