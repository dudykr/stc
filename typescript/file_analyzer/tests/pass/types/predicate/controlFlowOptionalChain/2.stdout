warning: Type
 --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:3:11
  |
3 | if (o2?.f(x)) {
  |           ^
  |
  = note: (string | number)

warning: Type
 --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:3:5
  |
3 | if (o2?.f(x)) {
  |     ^^
  |
  = note: ({
              f(x: any): x is number;
          } | undefined)

warning: Type
 --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:3:5
  |
3 | if (o2?.f(x)) {
  |     ^^^^^
  |
  = note: (undefined | (x: any) => x is number)

warning: Type
 --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:3:5
  |
3 | if (o2?.f(x)) {
  |     ^^^^^^^^
  |
  = note: x is number

warning: Type
 --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:4:5
  |
4 |     x; // number
  |     ^
  |
  = note: number

warning: Type
 --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:5:5
  |
5 |     o2.f; // (x: any) => x is number
  |     ^^
  |
  = note: {
              f(x: any): x is number;
          }

warning: Type
 --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:5:5
  |
5 |     o2.f; // (x: any) => x is number
  |     ^^^^
  |
  = note: (x: any) => x is number

warning: Type
 --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:6:5
  |
6 |     o2?.f;
  |     ^^
  |
  = note: {
              f(x: any): x is number;
          }

warning: Type
 --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:6:5
  |
6 |     o2?.f;
  |     ^^^^^
  |
  = note: (x: any) => x is number

warning: Type
 --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:7:11
  |
7 |     o2?.f(x);
  |           ^
  |
  = note: number

warning: Type
 --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:7:5
  |
7 |     o2?.f(x);
  |     ^^
  |
  = note: {
              f(x: any): x is number;
          }

warning: Type
 --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:7:5
  |
7 |     o2?.f(x);
  |     ^^^^^
  |
  = note: (x: any) => x is number

warning: Type
 --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:7:5
  |
7 |     o2?.f(x);
  |     ^^^^^^^^
  |
  = note: x is number

warning: Type
  --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:10:5
   |
10 |     x;
   |     ^
   |
   = note: string

warning: Type
  --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:11:5
   |
11 |     o2;
   |     ^^
   |
   = note: ({
               f(x: any): x is number;
           } | undefined)

warning: Type
  --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:12:5
   |
12 |     o2?.f;
   |     ^^
   |
   = note: ({
               f(x: any): x is number;
           } | undefined)

warning: Type
  --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:12:5
   |
12 |     o2?.f;
   |     ^^^^^
   |
   = note: (undefined | (x: any) => x is number)

warning: Type
  --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:13:5
   |
13 |     o2.f;
   |     ^^
   |
   = note: ({
               f(x: any): x is number;
           } | undefined)

warning: Type
  --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:13:5
   |
13 |     o2.f;
   |     ^^^^
   |
   = note: (x: any) => x is number

warning: Type
  --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:15:1
   |
15 | x;
   | ^
   |
   = note: (string | number)

warning: Type
  --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:16:1
   |
16 | o2;
   | ^^
   |
   = note: ({
               f(x: any): x is number;
           } | undefined)

warning: Type
  --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:17:1
   |
17 | o2?.f;
   | ^^
   |
   = note: ({
               f(x: any): x is number;
           } | undefined)

warning: Type
  --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:17:1
   |
17 | o2?.f;
   | ^^^^^
   |
   = note: (undefined | (x: any) => x is number)

warning: Type
  --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:18:1
   |
18 | o2.f;
   | ^^
   |
   = note: ({
               f(x: any): x is number;
           } | undefined)

warning: Type
  --> $DIR/tests/pass/types/predicate/controlFlowOptionalChain/2.ts:18:1
   |
18 | o2.f;
   | ^^^^
   |
   = note: (x: any) => x is number

